### PR TITLE
Send cookies explicitly into `context.request`'s transaction

### PR DESCRIPTION
**Pull request by david-luna**
# Description
This PR changes the transaction serialisation to be more explicit about sanitization of sensitive data in request's `cookie` header.

**Current status**

cookies are sanitized in the header value based on [sanitizeFieldNames](https://www.elastic.co/guide/en/apm/agent/nodejs/current/configuration.html#sanitize-field-names) configuration. As a result you see clear values and redacted ones. This may lead to think the agent is not properly doing sanitization.

**After this PR**

cookies are going to be placed in `context.request.cookies` map, which is accepted by the [intake API](https://github.com/elastic/apm-server/blob/d18c559e4136b2213a4f6e206258c5d303ec1b0f/docs/spec/v2/transaction.json#L129-L135), and sanitized based on the configuration. The cookie header will be completely redacted to avoid duplication of data.

Fixes: #3273 
### Checklist

- [x] Implement code
- [x] Add tests
- [ ] Update TypeScript typings
- [ ] Update documentation
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -21594,6 +21612,29 @@
       "integrity": "sha512-OPs5WnnT1xkCBiuQrZA4+YAV4HEJejmHneyraIaxsbev5yCEr6KMwINNFP9wQeFIw8FWcoTqF3vQsa5CDaI+8Q==",
       "dev": true
     },
+    "@types/node-fetch": {
```
[the remark this answers is not in the document]
Comment (trentm) [reply package-lock.json]: This should be from https://github.com/elastic/apm-agent-nodejs/pull/3318/files

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -68,6 +68,8 @@ Notes:
 [float]
 ===== Chores
 
+* Add cookie map in transactions' request context and redact cookie header ({pull}3304[#3322])
```
Comment (trentm) [inline CHANGELOG.asciidoc:71]:
```suggestion
* Add cookie map in transactions' request context and redact cookie header ({pull}3322[#3322])
```

File: lib/parsers.js [hunk 3]
```diff
@@ -58,6 +60,15 @@ function getContextFromRequest (req, conf, type) {
       Object.assign({}, req.headers),
       conf.sanitizeFieldNamesRegExp
     )
+
+    // TODO: deprecate filterHttpHeaders in next major release
```
Comment (trentm) [inline lib/parsers.js:64]: We might be disagreeing on the definition of "deprecate". For me, "deprecating" means that we would document (and mark in types) that the config var is "deprecated" -- hence users should avoid it and the docs should show users how to migrate away from it, but the field is still supported.

The "TODO" for the next major release would be to *remove* the `filterHttpHeaders` config var.
Thoughts?

If we agree, then would you like to do the deprecation tweaks in this PR, or prefer that we do it in a separate change?

**Review by trentm — APPROVED** [review]
One type on the changelog, and a question about the deprecation below. Otherwise this looks good.

File: lib/parsers.js (hunk 3 above)
Comment (david-luna) [reply lib/parsers.js:64]: I think your approach is much better. I shall do the changes in this PR.

Thanks!

**Pull request merged**